esphome/dashboard/web_server.py

```diff
@@ -725,8 +729,13 @@ async def get(self, configuration: str | None = None) -> None:
             self.send_error(404)
             return
 
-        data = await loop.run_in_executor(None, self._load_file, path, compressed)
-        self.write(data)
+        # MD5 files should not be compressed and are text files
+        if file_name.endswith(".md5"):
+            with open(path, encoding="utf-8") as f:
+                self.write(f.read())
+        else:
+            data = await loop.run_in_executor(None, self._load_file, path, compressed)
+            self.write(data)
```

Comment:
this feels even more needlessly complex. I think this should just stay original unless someone smarter than me thinks md5s needs their own codepath.
```suggestion
        data = await loop.run_in_executor(None, self._load_file, path, compressed)
        self.write(data)
```